internal/cli/klog.go

```diff
@@ -3,23 +3,73 @@ package cli
 import (
 	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
+	"regexp"
 
 	"k8s.io/klog"
 )
 
 var klogLevel = 0
 
+type filteredWriter struct {
+	underlying io.Writer
+	filterFunc func(s string) bool
+	leftover   []byte
+}
+
+func (fw *filteredWriter) Write(buf []byte) (int, error) {
+	buf = append(fw.leftover, buf...)
+	start := 0
+	written := 0
+	for i, b := range buf {
+		if b == '\n' {
+			end := i
+			if buf[i-1] == '\r' {
+				end--
+			}
+			s := string(buf[start:end])
+
+			if !fw.filterFunc(s) {
+				n, err := fw.underlying.Write(buf[start : i+1])
+				written += n
+				if err != nil {
+					fw.leftover = append([]byte{}, buf[i+1:]...)
+					return written, err
+				}
+			}
+
+			start = i + 1
+		}
+	}
+
+	fw.leftover = append([]byte{}, buf[start:]...)
+
+	return written, nil
+}
+
+// lines matching `filterFunc` will not be output to the underlying writer
+func newFilteredWriter(underlying io.Writer, filterFunc func(s string) bool) io.Writer {
+	return &filteredWriter{underlying: underlying, filterFunc: filterFunc}
+}
+
+// everything on google indicates this warning is useless and should be ignored
```

Comment:
Is there a bug upstream yet that we can link to? If not, can you file one?

 In the past, the kubernetes people have seemed pretty receptive to prs that add hooks for silencing warnings.